### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by JoshuaMulliken**
The purpose of this refactor is to reduce the amount of duplicate code and increase ability to fix bugs and add new devices and features as they come along.

It introduces the concept of IUpdatable and ISwitchable. These interfaces are used in the client.py and let the user choose what actions they want to add to their device. A call to update requires a device that implements IUpdatable. This allows the client to own all the knowledge about the API and the device to own the knowledge of itself.

Any recommendations are very welcome.

### CODE REVIEW

File: custom_components/wyzeapi/light.py [hunk 1]
```diff
@@ -94,12 +109,12 @@ def brightness(self):
...
         that brightness is not supported for this light.
         """
-        return self.__light.brightness
+        return self.translate(self.__light.brightness, 1, 100, 1, 255)
 
     @property
     def color_temp(self):
         """Return the CT color value in mired."""
-        return self.__light.color_temp
+        return self.translate(self.__light.color_temp, 2700, 6500, 500, 140)
```
Comment (markrickert) [inline custom_components/wyzeapi/light.py:117]: There's a lot of repeating integers hanging around in this file. Maybe break it out into functions `translate_brightness` and `translate_color_temp`? However, it appears that this color temp translation params are different from the ones below on line 135... is this intentional?

File: custom_components/wyzeapi/wyzeapi/devices/base_sensor.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        if self.product_model == "PIR3U":
+            prop_map.update({
+                "P1302": self.open_close_state
+            })
+        if self.product_model == "DWS3U":
+            prop_map.update({
+                "P1301": self.open_close_state
+            })
+
```
Comment (markrickert) [inline custom_components/wyzeapi/wyzeapi/devices/base_sensor.py:33]: This is much more maintainable!

File: custom_components/wyzeapi/wyzeapi/devices/lock.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+from typing import Dict
+
+from .base import BaseDevice
+from ..interfaces import IUpdatable
+
+
```
Comment (markrickert) [inline custom_components/wyzeapi/wyzeapi/devices/lock.py:6]: > TODO: Add support for locking and unlocking, not just reading state.

Comment (markrickert) [reply custom_components/wyzeapi/wyzeapi/devices/lock.py:6]: 🤣

**Comment by jfarmer08** [comment]
You have done great work with this. With this update it will make it a lot easier to manage. I don't see anything that stands out.

File: custom_components/wyzeapi/light.py (hunk 1 above)
Comment (JoshuaMulliken) [reply custom_components/wyzeapi/light.py:117]: Yeah it is. It makes sure the changes go both ways.

**Pull request merged**